mayastor/src/bdev/nexus/nexus_noddy_rebuild.rs

```diff
@@ -0,0 +1,90 @@
+//!
+//! This file implements operations for the rebuild process
+//!
+//! `noddy_rebuild` does a noddy rebuild
+//! 
+
+use crate::{
+    bdev::nexus::{
+        nexus_bdev::{
+            Nexus,
+            NexusState,
+        },
+        nexus_child::{ChildState},
+        nexus_channel::DREvent,
+    },
+};
+
+use std::cmp;
+
+impl Nexus {
+    // rebuilds only the first bad child it finds
+    pub(crate) async fn noddy_rebuild(&mut self) -> NexusState {
+        // if failed go back to the previous state
+        let state = self.state;
+
+        self.set_state(NexusState::Remuling);
+
+        let good_child = match self.children.iter().find(|c| c.repairing == false) {
+            Some(good_child) => good_child,
+            None => return self.set_state(state),
+        };
+
+        let bad_child = match self.children.iter().find(|c| c.repairing == true) {
+            Some(bad_child) => bad_child,
+            None => return self.set_state(state),
+        };
+
+        let bdev_handle = match &bad_child.bdev_handle {
+            Some(bdev_handle) => bdev_handle,
+            None => return self.set_state(state),
+        };
+
+        let block_size = self.bdev.block_len() as u64;
+        let max_rebuild_size = 10u64 * 1024 * 1024; // 10MiB
+        let max_blocks = max_rebuild_size / block_size;
+
+        // At the moment if we copy the whole bdev it causes the js tests to timeout
+        // One option is to invoke the rebuild through a different command (ai - not on the add_child)
```

Comment:
Did you mean "i.e. - not on the add_child"?

Reply:
Done